certification/engine/engine.go

```diff
@@ -174,3 +181,25 @@ func ScratchContainerPolicy() []string {
 func RootExceptionContainerPolicy() []string {
 	return makeCheckList(rootExceptionContainerPolicy)
 }
+
+func getClient() crclient.Client {
+	scheme := apiruntime.NewScheme()
+	operatorv1.AddToScheme(scheme)
+	operatorv1alpha1.AddToScheme(scheme)
+	imagestreamv1.AddToScheme(scheme)
+	kubeconfig, err := ctrl.GetConfig()
+	if err != nil {
+		log.Error("could not get kubeconfig")
+		return nil
+	}
+
+	client, err := crclient.New(kubeconfig, crclient.Options{
+		Scheme: scheme,
+	})
+	if err != nil {
+		log.Error("could not get operator group client")
+		return nil
+	}
+
+	return client
+}
```

Comment:
This introduces as bit of a UX problem. We initialize our checks right out of the gate of Preflight execution. That means that this code is getting called regardless of what subcommand a user is running. The result is this:

```
$ ./preflight --version
ERRO[0000] could not get operator group client          
preflight version 0.0.0 <commit: 6e773adc8716ab20685474f2576f61bfd9732271>
```

That's because I have KUBECONFIG unset. If I have it set, I still see:

```
$ KUBECONFIG="$PWD/KUBECONFIG" ./preflight --version
I0516 13:40:33.943636    4469 request.go:665] Waited for 1.016926668s due to client-side throttling, not priority and fairness, request: GET:https://mycluster:6443/apis/quota.openshift.io/v1?timeout=32s
preflight version 0.0.0 <commit: 6e773adc8716ab20685474f2576f61bfd9732271>
```

I'm not sure if we want to look at lazy loading the client here to resolve that? Or maybe make this work the first thing that happens when the check runs? 

-

Secondly, a tiny nit, and definitely less importantly, the use of `get` here, and that being an "antipattern"[1].

Given that we're not really calling this on a struct, I don't care if you leave it this way or otherwise.

[1]
https://go.dev/doc/effective_go#Getters

Reply:
This is precisely the piece of code that I wanted eyes on the most. Thanks for the feedback. Clearly this needs more thought because of the loading concern.

Reply:
The solution here might be something like... accept a `*crclient.Client*`, pass it `nil`, and then run an initializer method on `Validate()` that sets up the client at that time. It's dirty, but it should push the initialization later so it doesn't impact cases that it should not be impacting.

Reply:
Thanks. I will play with that.